### DISCUSSION TITLE
Limit initial snapshot

### DISCUSSION THREAD

**Discussion by mwisnicki** [MaterializeInc/materialize#30093] [Q&A] [answered]
Is it possible to limit or skip initial snapshot with PostgreSQL connector?

I want to quickly test materialize with massive database and don't need most existing data, just real-time changes.

**Comment by morsapaes** [comment, marked as answer]
This is not possible, since we need to take a consistent snapshot of the existing database state before resuming real-time ingestion. For testing, we recommend limiting the number of tables you add to the publication by using the `CREATE PUBLICATION...FOR TABLE` (instead of `FOR ALL TABLES`) clause (as described in the [documentation](https://materialize.com/docs/ingest-data/postgres/self-hosted/#2-create-a-publication-and-a-replication-user)).
